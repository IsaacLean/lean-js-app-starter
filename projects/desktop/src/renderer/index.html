<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <%= csp %>
    <title>Lean JavaScript Application Starter</title>
  </head>
  <body>
    <div id="root"></div>
    <noscript>You must enable JavaScript to use this application!</noscript>
  </body>
</html>