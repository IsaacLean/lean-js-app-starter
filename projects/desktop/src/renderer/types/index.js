// @flow

export type Action =
  | Action_Decrement
  | Action_Increment

export type Action_Decrement = {|
  type: 'DECREMENT'
|}

export type Action_Increment = {|
  type: 'INCREMENT'
|}

export type State = {|
  +counter: State_Counter
|} | {||}

export type State_Counter = number
