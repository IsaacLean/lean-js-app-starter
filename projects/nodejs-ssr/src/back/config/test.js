// @flow

export default {
  DB_URI: 'mongodb://localhost/nodejs-ssr-test'
}
