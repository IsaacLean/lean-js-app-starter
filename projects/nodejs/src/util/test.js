// @flow

const logger = require('./logger')

const test = {
  clearDBCollection(model: string) {
    if(process.env.NODE_ENV === 'test') {
      const Model = require(`../models/${model}`)

      return Model.remove({})
        .exec()
        .catch(err => logger.error(err))
    }

    throw new Error('Invalid environment')
  }
}

module.exports = test
