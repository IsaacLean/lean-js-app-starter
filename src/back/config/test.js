// @flow

export default {
  DB_URI: 'mongodb://localhost/starchan-test'
}
