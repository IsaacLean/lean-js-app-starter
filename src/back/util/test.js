// @flow

import logger from './logger'

const test = {
  clearDBCollection(model: string) {
    if(process.env.NODE_ENV === 'test') {
      const Model = require(`../models/${model}`).default

      return Model.remove({})
        .exec()
        .catch(err => logger.error(err))
    }

    throw new Error('Invalid environment')
  }
}

export default test
