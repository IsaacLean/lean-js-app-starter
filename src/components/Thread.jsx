// @flow
import React from 'react'
import moment from 'moment'

import Reply from './Reply'
import type { Thread as ThreadType } from '../types'

const Thread = ({ data }: { data: ThreadType }) => {
  let subject
  if(data.subject) {
    subject = <strong>{data.subject} {' '}</strong>
  }

  let replies
  if(Array.isArray(data.replies) && data.replies.length > 0) {
    replies = data.replies.map(reply => <Reply key={reply._id} data={reply} />)
    replies = <ul>{replies}</ul>
  }

  return <li className="thread">
    <header>
      {subject}
      <b className="name">Anonymous</b>
      {' '}
      {moment(data.createdAt).format('MM/DD/YY(ddd)HH:DD:SS')}
      {' Id.'}
      {data._id}
    </header>
    <section className="comment">
      {data.comment}
    </section>
    {replies}
  </li>
}

export default Thread
