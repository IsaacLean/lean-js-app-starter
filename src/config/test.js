// @flow

module.exports = {
  DB_URI: 'mongodb://localhost/starchan-test'
}
