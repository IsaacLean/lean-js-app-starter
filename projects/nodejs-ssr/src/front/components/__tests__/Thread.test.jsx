import React from 'react'
import renderer from 'react-test-renderer'
import { HashRouter as Router } from 'react-router-dom'

import Thread from '../Thread'
import { threads } from '../../util/test/data'

describe('Thread', () => {
  it('renders properly', () => {
    const thread = threads[0]
    const component = renderer.create(
      <Router>
        <Thread data={thread} />
      </Router>
    )
    expect(component).toMatchSnapshot()
  })
})
