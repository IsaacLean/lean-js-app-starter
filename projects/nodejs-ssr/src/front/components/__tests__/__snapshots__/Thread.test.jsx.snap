// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thread renders properly 1`] = `
<li
  className="thread"
>
  <header>
    <strong>
      New Thread
       
       
    </strong>
    <b
      className="name"
    >
      Anonymous
    </b>
     
    11/15/18(Thu)18:15:30
     Id.
    5bee2f90c43e83433b5e5d24
     
    [
    <a
      href="#/5bee2f90c43e83433b5e5d24"
      onClick={[Function]}
    >
      Reply
    </a>
    ]
  </header>
  <pre
    className="comment"
  >
    I am the new thread comment.
  </pre>
  <ul>
    <li
      className="reply"
    >
      <div>
        <header>
          <b
            className="name"
          >
            Anonymous
          </b>
           
          11/15/18(Thu)18:15:31
           Id.
          5bee2f90c43e83433b5e5d25
        </header>
        <pre
          className="comment"
        >
          All your base are belong to us.
        </pre>
      </div>
    </li>
  </ul>
</li>
`;
